File: tests/wasm2lang_08_factorial.harness.mjs
'use strict';

const moduleImports = {};

const runTest = function (buff, out, exports, data) {
  exports.alignHeapTop();

  for (const v of data.factorial_inputs) {
    exports.exerciseFactorial(v);
  }
};

const dumpMemory = true;

export { dumpMemory, moduleImports, runTest };

File: tests/wasm2lang_08_factorial.harness.java
{
    WasmModule mod = new WasmModule(new java.util.LinkedHashMap<>(), memBuffer);
    mod.alignHeapTop();

    java.util.Map<String, Object> _data = w2lLoadSharedData(System.getProperty("w2l.testname", ""));

    for (Double v : w2lFlat(_data, "factorial_inputs")) {
        mod.exerciseFactorial(v.intValue());
    }

    w2lDumpCRC(memBuffer);
}

/exit
